src/main/java/com/amogoscode/groupe/ebankingsuite/User/UserService.java

```diff
@@ -0,0 +1,62 @@
+package com.amogoscode.groupe.ebankingsuite.User;
+
+import com.amogoscode.groupe.ebankingsuite.User.resquests.UserAuthenticationRequests;
+import com.amogoscode.groupe.ebankingsuite.User.resquests.UserRegistrationRequest;
+import com.amogoscode.groupe.ebankingsuite.config.JWTService;
+import com.amogoscode.groupe.ebankingsuite.exception.InvalidAuthenticationException;
+import com.amogoscode.groupe.ebankingsuite.exception.ResourceNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import javax.naming.AuthenticationException;
+import java.util.Optional;
+
+@Service
+public class UserService {
+    private final UserRepository userRepository;
+    private final JWTService jwtService;
+    private static final BCryptPasswordEncoder bCryptPasswordEncoder =
+            new BCryptPasswordEncoder();
+
+    public UserService(UserRepository userRepository, JWTService jwtService) {
+        this.userRepository = userRepository;
+        this.jwtService = jwtService;
+    }
+
+    //this service creates a new user provided the email does not exist
+    //later this method will be updated to invoke the create user account service to create an account for the user
+    public void createNewUser(UserRegistrationRequest userRegistrationRequest){
+        if(!userRepository.existsByEmailAddress(userRegistrationRequest.emailAddress())){
```

Comment:
it's not a best practice to do return; in void functions
instead you can check the opposite condition and throw and error, and if the condition not met, then save it without return;

Reply:
This will be resolved and looked out for moving forward

Reply:
changes have been pushed in the new commit